ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java

```diff
@@ -152,17 +152,18 @@ public void writeRecord(final ConsumerRecord<byte[], byte[]> record) {
 
   void writeCommandToBackup(final ConsumerRecord<CommandId, Command> record) {
     if (corruptionDetected) {
-      LOG.warn("Failure to write command topic data to backup. "
-          + "Corruption detected in command topic.");
-      return;
+      throw new KsqlServerException(
+          "Failed to write record due to out of sync command topic and backup file: " + record);
     }
 
     if (isRestoring()) {
       if (isRecordInLatestReplay(record)) {
         // Ignore backup because record was already replayed
         return;
       } else {
-        LOG.info("Previous command topic backup does not match the new command topic data.");
+        LOG.warn("Backup is out of sync with the current command topic. "
+            + "Backups will not work until the previous command topic is "
+            + "restored or all backup files are deleted.");
```

Comment:
Throw the exception here, otherwise the first time it will only log.

Reply:
Btw, do not log this message once you throw the exception. The log should be done by the method who catches the exception.

Reply:
done